libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/api/ElasticsearchEntitlementChecker.java

```diff
@@ -305,4 +312,68 @@ public ElasticsearchEntitlementChecker(PolicyManager policyManager) {
     public void check$javax_net_ssl_SSLContext$$setDefault(Class<?> callerClass, SSLContext context) {
         policyManager.checkChangeJVMGlobalState(callerClass);
     }
+
+    @Override
+    public void check$java_net_DatagramSocket$bind(Class<?> callerClass, DatagramSocket that, SocketAddress addr) {
+        policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.LISTEN_ACTION);
+    }
+
+    @Override
+    public void check$java_net_DatagramSocket$connect(Class<?> callerClass, DatagramSocket that, InetAddress addr) {
+        policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.CONNECT_ACTION | NetworkEntitlement.ACCEPT_ACTION);
+    }
+
+    @Override
+    public void check$java_net_DatagramSocket$connect(Class<?> callerClass, DatagramSocket that, SocketAddress addr) {
+        policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.CONNECT_ACTION | NetworkEntitlement.ACCEPT_ACTION);
+    }
+
+    @Override
+    public void check$java_net_DatagramSocket$send(Class<?> callerClass, DatagramSocket that, DatagramPacket p) {
+        var actions = NetworkEntitlement.CONNECT_ACTION;
+        if (p.getAddress().isMulticastAddress()) {
+            actions |= NetworkEntitlement.ACCEPT_ACTION;
+        }
+        policyManager.checkNetworkAccess(callerClass, actions);
+    }
+
+    @Override
+    public void check$java_net_DatagramSocket$receive(Class<?> callerClass, DatagramSocket that, DatagramPacket p) {
+        policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.ACCEPT_ACTION);
```

Comment:
Conversely, this is the only place where `ACCEPT_ACTION` is used without `CONNECT_ACTION`.

It makes me wonder if we actually need to specify these separately. 🤔

Reply:
I wonder if we really need just two kinds of actions for `INCOMING` and `OUTGOING` network traffic. 🤔

Reply:
Correct. 
They are basically representing in and out connections; that said, I'd rather keep the accept/connect/listen "verbs" as they are, they make a bit more sense/are a bit more "familiar" in relation to network operations.